jaxrs-service/src/main/java/com/quorum/tessera/admin/ConfigResource.java

```diff
@@ -85,4 +90,49 @@ public Response getPeers() {
         }).build();
     }
 
+    @GET
+    @Path("/keypairs/{publicKey}")
+    public Response getKeyPair(@PathParam("publicKey") String base64PublicKey) {
+
+        Base64.Decoder base64Decoder = Base64.getDecoder();
+
+        PublicKey publicKey = PublicKey.from(base64Decoder.decode(base64PublicKey));
+
+        Set<PublicKey> publicKeys = configService.getPublicKeys();
+
+        if(!publicKeys.contains(publicKey)) {
+            throw new NotFoundException("No key pair found with public key " + base64PublicKey);
+        }
+
+        //TODO use dedicated transport object instead?
+        KeyData responseData = new KeyData();
+        responseData.setPublicKey(base64PublicKey);
+        responseData.setPrivateKey("REDACTED");
+
+        return Response.ok(responseData).build();
+    }
+
+    @GET
+    @Path("/keypairs")
+    public Response getKeyPairs() {
+        Set<PublicKey> publicKeys = configService.getPublicKeys();
+
+        if(publicKeys.isEmpty()) {
```

Comment:
It would be better for return the empty list instead of throwing an error here; the resource exists, but is just empty.